### PR TITLE
Fix visually hidden classnames

### PR DESCRIPTION
**Pull request by talldan**
## Description
While reviewing #20607, I noticed `VisuallyHidden` doesn't support a `className` prop. If `className` is specified, the existing `components-visually-hidden` class is overwritten, making the component no longer hidden.

There was a use case where spreading props onto `VisuallyHidden` seemed like a reasonable option (https://github.com/WordPress/gutenberg/pull/20607#discussion_r388148496), so I thought this would be an opportunity to change the component to better support class names.

I think there are two options:
1. Class names received through the `className` prop are appended.
2. Class names received through the `className` prop are ignored.

I went with the first option for consistency with other components, but I think the second also has merit—it'd mean other styles couldn't break `VisuallyHidden`. Happy to change the approach based on feedback.

## How has this been tested?
Added snapshot tests.

For manual testing:
1. Find a VisuallyHidden component
2. Add a className prop
3. Observe that the class name is appended to the components class name.

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/components/src/visually-hidden/index.js [hunk 1]
```diff
@@ -7,10 +12,10 @@ import { renderAsRenderProps } from './utils';
  * VisuallyHidden component to render text out non-visually
  * for use in devices such as a screen reader.
  */
-function VisuallyHidden( { as = 'div', ...props } ) {
+function VisuallyHidden( { as = 'div', className, ...props } ) {
```
Comment (gziolo) [inline packages/components/src/visually-hidden/index.js]: nice 👍

File: packages/components/src/visually-hidden/test/index.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ */
+import TestRenderer from 'react-test-renderer';
+
+/**
+ * Internal dependencies
+ */
+import VisuallyHidden from '../';
+
+describe( 'VisuallyHidden', () => {
+	it( 'by default renders as a div with the visually hidden class name', () => {
```
Comment (gziolo) [inline packages/components/src/visually-hidden/test/index.js]: Two first tests duplicate unit tests generated from Storybook stories. See:

https://github.com/WordPress/gutenberg/blob/master/packages/components/src/visually-hidden/stories/index.js

https://github.com/WordPress/gutenberg/blob/cf7f028c270ccabb19e3f2fca7e5098379b0dc6e/storybook/test/__snapshots__/index.js.snap#L8773-L8791

Do you think we should add more stories instead, or should those tests be more specif in how verification happens?

**Comment by tellthemachines** [comment]
> I went with the first option for consistency with other components, but I think the second also has merit—it'd mean other styles couldn't break `VisuallyHidden`. Happy to change the approach based on feedback.

Great point! I think the first approach could be useful if we ever want to hide/unhide something with a transition, like a skip link for instance. Or maybe that scenario would make more sense as a different component? 
Other than that, agree that the second approach is more fail-proof 😄

File: packages/components/src/visually-hidden/test/index.js (hunk 2 above)
Comment (talldan) [reply packages/components/src/visually-hidden/test/index.js]: Ah, I didn't actually know stories are snapshot tested! Makes absolute sense to use stories for these tests. I'll update the PR.

**Review by gziolo — APPROVED** [review]
Nice, let's give it a try. If we discover issues we can always revert it 😃

**Pull request merged**